pkg/controller/subscription/rhmiConfigs/main.go

```diff
@@ -73,6 +85,8 @@ func UpdateStatus(ctx context.Context, client k8sclient.Client, config *integrea
 		config.Status.Maintenance.Duration = strconv.Itoa(WINDOW) + "hrs"
 	}
 
+	config.Status.TargetVersion = targetVersion
+
 	// Calculate the upgrade window
 	if installplan.Spec.Approved {
 		config.Status.Upgrade.Window = ""
```

Comment:
```suggestion
		config.Status.Upgrade.Window = ""
		config.Status.TargetVersion = ""
```
I'd clear the targetVersion field once the upgrade is approved